TDex-network/tdex-daemon
# Fix wallet restore

This starts from #312 and fixes the restoration of the wallet.

@tiero do you want to test this out?

I made several tries to check if all works as expected. For each of the following situations I tried to complete the flow in order to broadcast at least 2 trades (1 BUY and 1 SELL):
* new wallet from scratch
* restart (start-stop-start) with already used wallet 
* restore already used wallet
* restart from restored wallet (start-with-restore-stop-start)

This makes the HTTP request timeout configurable while previously was hardcoded to 3 secs and was cause of problems when using Esplora explorer.

Also, this introduces the usage of circuit breaker pattern to retry failing HTTP requests when fetching/restoringthe utxo set.


~~What I noticed:
sometimes (but it might actually  be everytime), with a new wallet from scratch, the TradePropose RPC fails with some `Invalid prefix...` error. When that happens I just restart the daemon so it re-builds the utxo set and everything works fine then. Would be nice if someone replicated this. I suspect the problem could come from the ClaimDeposit RPC(s)~~

EDIT: fixed.
